### PR TITLE
Handle exit and shutdown request

### PR DESCRIPTION
**Pull request by faustinoaq**
Fixes https://github.com/crystal-lang-tools/scry/issues/69 and replaces #70 

/cc @damieng

### CODE REVIEW

**Comment by faustinoaq** [comment]
#75 already cover an spec for this :+1:

File: src/scry/context.cr [hunk 1]
```diff
@@ -27,26 +29,32 @@ module Scry
...
       Log.logger.debug(msg.method)
-      dispatchNotification(msg.params, msg)
+      dispatch_notification(msg.params, msg)
+    end
+
+    private def dispatch_request(params : Nil, msg)
+      if msg.method == "shutdown"
+        Scry.shutdown = true
+        ResponseMessage.new(msg.id, nil)
+      elsif msg.method == "exit"
```
Comment (damieng) [inline src/scry/context.cr]: I think the 'exit' handling needs to be in dispatch_notification as it is a notification message not a request message (unlike shutdown which is in the correct place)

Comment (faustinoaq) [reply src/scry/context.cr]: Oh, yeah, Thank you!

**Comment by damieng** [comment]
Looks good to me.

**Pull request merged**